consuming-webservices/Networking/RequestHandler.swift

```diff
@@ -45,12 +45,44 @@ struct HTTPRequestHandler: RequestHandler {
     var body: Any?
     
     func execute( callback: @escaping (Result<Any>) -> Void) {
-        var returnValue = 0
-        if let body = body as? [String: String] {
-            returnValue = body["args"] == "on" ? 1 : 0
+        guard let url = URL(string: path) else { return }
+        var request = URLRequest(url: url)
+        request.httpMethod = method.rawValue
+        
+        if let headers = headers {
+            if let contentType = headers["Content-Type"] {
+                request.addValue(contentType, forHTTPHeaderField: "Content-Type")
+            }
+        }
+        
+        if let body = body {
+            do {
+                request.httpBody = try JSONSerialization.data(withJSONObject: body, options: [])
+            } catch (let e) {
+                callback(.failure(e))
+            }
         }
         
-        let fakeData = ["return_value": returnValue, "result": 0]
-        callback(.success(fakeData))
+        let task = URLSession.shared.dataTask(with: request) { data, response, error in
+            if let error = error {
+                callback(.failure(error))
+                return
+            }
+            guard let data = data else {
+                callback(.failure(RequestError.noData))
+                return
+            }
+            
```

Comment:
Ideally you'd do something with the `response`, since it can give you data like "404: not found" or "503: service unavailable"